src/deadlock-detection/lockorder.cpp

```diff
@@ -158,4 +158,19 @@ void CLockOrderTracker::TrackLockOrderHistory(const CLockLocation &locklocation,
     }
 }
 
+void CLockOrderTracker::DeleteCritical(void *cs)
+{
+    std::map<void *, std::set<void *> >::iterator iter;
```

Comment:
don't you need to lock here at the beginning of DeleteCritical?

  std::lock_guard<std::mutex> lock(lot_mutex);

Reply:
This DeleteCritical is only called within the lock taken by the global DeleteCritical function, so the lock is taken.  I would like to see a _ prefix in the name and an assert lock held call at the top. But I am going to merge this to move forward with testing.

Reply:
the locks that belong to the detection code itself are not tracked, so you cannot call assertlockheld on these locks.

ptschip is right, it would be correct to lock the lot_mutex here. i will fix that in a future PR.